src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+
+document.getElementById('header-logo').src = logo;
+document.getElementById('profile-avatar').src = avatar;
+document.getElementById('profile-post-icon').src = plusIcon;
+document.getElementById('profile-avatar-edit-icon').src = lightPencilIcon;
+document.getElementById('profile-edit-icon').src = pencilIcon;
+
+const api = new Api({
+  baseUrl: 'https://around-api.en.tripleten-services.com/v1',
+  headers: {
+      authorization: 'd9b462f6-0ab1-4657-9af2-202693166e25',
+      'Content-Type': 'application/json'
+  }
+});
+
+
+
+  // Edit Profile Modal
+  const profileName = document.querySelector('.profile__name');
+  const profileDescription = document.querySelector('.profile__description');
+  const editButton = document.querySelector('.profile__edit-btn');
+  const editProfileModal = document.querySelector('#edit-profile');
+  const editProfileCloseButton = editProfileModal.querySelector('#edit-profile-close-btn');
+  const editModalNameInput = editProfileModal.querySelector('#profile-name-input');
+  const editModalDescriptionInput = editProfileModal.querySelector('#profile-description-input');
+  const editProfileForm = document.forms['profile-form'];
+
+  // New Post Modal
+  const newPostButton = document.querySelector('.profile__post-btn');
+  const newPostModal = document.querySelector('#add-card-modal');
+  const newPostCloseButton = newPostModal.querySelector('#new-post-close-btn');
+  const newPostForm = document.forms['new-post'];
+  const newPostLinkInput = newPostModal.querySelector('#add-card-link-input');
+  const newPostCaptionInput = newPostModal.querySelector('#add-card-caption-input');
+  const cardSubmitBtn = newPostModal.querySelector(".modal__submit-btn");
+
+  // Preview Modal
+  const previewModal = document.querySelector('#preview-modal');
+  const previewModalImage = previewModal.querySelector('.modal__image');
+  const previewModalCaption = previewModal.querySelector('.modal__caption');
+  const previewCloseButton = previewModal.querySelector('.modal__close-btn_type_preview');
+
+  //Avatar Modal
+  const editAvatarModal = document.getElementById('edit-avatar-modal');
+  const editAvatarCloseButton = document.querySelector('#edit-avatar-close-btn');
+  const editAvatarForm = editAvatarModal.querySelector('form');
+  const editAvatarLinkInput = editAvatarModal.querySelector('#edit-avatar-link-input');
+  const avatarSubmitBtn = editAvatarModal.querySelector('.modal__submit-btn');
+  const profileAvatar = document.getElementById('profile-avatar');
+
+  function closeAvatarModal() {
+    closeModal(editAvatarModal);
+  }
+
+  const openAvatarModal = () => {
+    resetValidation(editAvatarForm, [editAvatarLinkInput]);
+    openModal(editAvatarModal);
+  };
+
+
+  function handleEditAvatarSubmit(evt) {
+    evt.preventDefault();
+
+    if (!editAvatarForm.checkValidity()) {
+      return;
+    }
+
+    const avatarUrl = editAvatarLinkInput.value;
+
+    avatarSubmitBtn.textContent = 'Saving...';
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions